csharp/ql/test/library-tests/ir/ir/raw_ir.expected

```diff
@@ -603,6 +603,58 @@ isexpr.cs:
 #   18|     v6_0(Void) = NoOp : 
 #-----|   Goto -> Block 1
 
+lock.cs:
+#    5| System.Void LockTest.A()
+#    5|   Block 0
+#    5|     v0_0(Void)            = EnterFunction              : 
+#    5|     mu0_1(null)           = AliasedDefinition          : 
+#    5|     mu0_2(null)           = UnmodeledDefinition        : 
+#    7|     r0_3(glval<Object>)   = VariableAddress[_object]   : 
+#    7|     r0_4(Object)          = NewObj                     : 
+#    7|     r0_5(glval<null>)     = FunctionAddress[Object]    : 
+#    7|     v0_6(Void)            = Call                       : func:r0_5, this:r0_4
+#    7|     mu0_7(null)           = ^CallSideEffect            : ~mu0_2
+#    7|     mu0_8(Object)         = Store                      : &:r0_3, r0_4
+#    8|     r0_9(glval<Object>)   = VariableAddress[#temp8:9]  : 
```

Comment:
One thing to note here is that both temp variables have the same name, because of how their names are created and I didn't want to modify the shared files that deal with this without consulting with the C++ team. What would be the best way to fix this ambiguity? What I was thinking was that we could have a new type of temp for compiler generated variables, whose name would be something like `#compiler_temp_uniqueID`;